spanner/cloud-client/src/main/java/com/example/spanner/AsyncQueryExample.java

```diff
@@ -0,0 +1,107 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+class AsyncQueryExample {
+
+  static void asyncQuery() throws InterruptedException, ExecutionException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String instanceId = "my-instance";
+    String databaseId = "my-database";
+
+    try (Spanner spanner =
+        SpannerOptions.newBuilder().setProjectId(projectId).build().getService()) {
+      DatabaseClient client =
+          spanner.getDatabaseClient(DatabaseId.of(projectId, instanceId, databaseId));
+      asyncQuery(client);
+    }
+  }
+
+  // Execute a query asynchronously and process the results in a callback.
+  static void asyncQuery(DatabaseClient client)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    ApiFuture<Void> finished;
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    try (AsyncResultSet resultSet =
+        client
+            .singleUse()
+            .executeQueryAsync(Statement.of("SELECT SingerId, AlbumId, AlbumTitle FROM Albums"))) {
+      // Setting a callback will automatically start the iteration over the results of the query
+      // using the specified executor. The callback will be called at least once. The returned
+      // ApiFuture is done when the callback has returned DONE and all resources used by the
+      // AsyncResultSet have been released.
+      finished =
+          resultSet.setCallback(
+              executor,
+              new ReadyCallback() {
+                @Override
+                public CallbackResponse cursorReady(AsyncResultSet resultSet) {
+                  try {
+                    while (true) {
+                      switch (resultSet.tryNext()) {
+                        // OK: There is a row ready.
+                        case OK:
+                          System.out.printf(
+                              "%d %d %s%n",
+                              resultSet.getLong(0), resultSet.getLong(1), resultSet.getString(2));
+                          break;
+
+                        // DONE: There are no more rows in the result set.
+                        case DONE:
+                          return CallbackResponse.DONE;
+
+                        // NOT_READY: There are currently no more rows in the buffer.
+                        case NOT_READY:
+                          return CallbackResponse.CONTINUE;
+
+                        default:
+                          throw new IllegalStateException();
+                      }
+                    }
+                  } catch (SpannerException e) {
+                    System.out.printf("Error in callback: %s%n", e.getMessage());
+                    return CallbackResponse.DONE;
+                  }
```

Comment:
This looks wrong style wise, but if lint is ok w/ it, I am as well.

I'm not usually a fan of catch / system.out style patterns, so, is there anything more to say here?  If not, it's ok.